src/bsoncxx/document/value.cpp

```diff
@@ -48,11 +48,43 @@ value& value::operator=(const value& rhs) {
     return *this;
 }
 
+view::const_iterator value::cbegin() const {
+    return this->view().cbegin();
+}
+
+view::const_iterator value::cend() const {
+    return this->view().cend();
+}
+
+view::const_iterator value::begin() const {
+    return this->view().begin();
+}
+
+view::const_iterator value::end() const {
+    return this->view().end();
+}
+
+view::const_iterator value::find(stdx::string_view key) const {
+    return this->view().find(key);
+}
+
 element value::operator[](stdx::string_view key) const {
     auto view = this->view();
     return view[key];
 }
 
+const std::uint8_t* value::data() const {
+    return static_cast<uint8_t*>(_data.get());
+}
+
+std::size_t value::length() const {
+    return _length;
+}
+
+bool value::empty() const {
+    return _length == 5;
+}
```

Comment:
These functions don't make a call to `view()`. I did this to save the driver from having to create a view to return information about the member fields it already has. Would it be better practice to have these functions call `return this->view().data()` instead to keep things consistent?

Reply:
Nope! This looks right. `value` really shouldn't depend on view for anything, but for it's simpler, for now, to call a `view` method. In fact, you should probably change `begin` and `end` to call `cbegin` and `cend` respectively the way `view` does it. That way, we reduce dependencies between the two.